jmxtrans-core/src/test/java/com/googlecode/jmxtrans/model/ServerTests.java

```diff
@@ -302,8 +302,6 @@ public void testConnectionRepoolingSkippedOnError_andConnectionIsClosed() throws
 			}
 		}
 
-		verify(conn).close();
```

Comment:
Shouldn't the connection be closed on invalidation? It's been a long time since I digged in this mess, but shouldn't pool.invalidate() destroy the connection?

Reply:
May be I don't understand what you mean.
This test using mocks. So we have check `orderVerifier.verify(pool).invalidateObject(server, conn);`. That's enough. Connection.close will call code inside pool.invalidateObject. It will call _factory.destroyObject(key, obj); which actually calls MBeanServerConnectionFactory.destroyObject which calls obj.close();

I wrote same in commit comment. You can check this call chain.

Reply:
Right! Sorry, my bad...